pkg/fatfs/fatfs_vfs/fatfs_vfs.c

```diff
@@ -47,6 +50,27 @@ static void _build_abs_path(fatfs_desc_t *fs_desc, const char *name)
              fs_desc->vol_idx, name);
 }
 
+#ifdef MODULE_FATFS_VFS_FORMAT
+static int _format(vfs_mount_t *mountp)
+{
+    fatfs_desc_t *fs_desc = mountp->private_data;
+    char volume_str[TEST_FATFS_MAX_VOL_STR_LEN];
+
+    BYTE *work = malloc(FF_MAX_SS);
```

Comment:
Do we really need `malloc` here?

Reply:
`FF_MAX_SS` is 512 byte

Reply:
The general (or maybe my strict?) expectation of RIOT functions without any admonitions is that they don't use the heap, are reentrant / thread safe, and don't overexert the stack. This can't hold that, so we should put a warning around the general [vfs_format](http://doc.riot-os.org/group__sys__vfs.html#ga85c6c8cab631ccd9ad61badb54d8ba10) function. So what'd that best be?

* "Not thread safe! Do not call this concurrently on different (let alone the same) file system.": Similar to the warning already on [vfs_iterate_mounts](http://doc.riot-os.org/group__sys__vfs.html#gafdbefec3679da1fed038f1fc42e8e821); would give license to use a static buffer.

  Advantages: Consistent with other vfs functions; memory limitations are visible (with `nm` showing the culprit) at build time if no heap allocation is used.

* "On some file systems (currently \\ref FatFS), this may resort to dynamic memory allocation, and return -ENOMEM if insufficient memory is available." This would justify the current behavior.

  Advantage: Works well on otherwise idle systems even with little memory.

* "Beware that some file systems (currently \\ref FatFS) have extensive stack requirements when running this". Very vague, and has bad error behavior.

Based on this, I'm softly leaning toward a const allocation.

Reply:
How about making it configurable? Allowing a static buffer to be provided, otherwise allocating one?

Reply:
I added `CONFIG_FATFS_FORMAT_ALLOC_STATIC`

Reply:
Is this a good enough compromise @bergzand? (I would personally prefer that the default be avoiding malloc though)

Reply:
After letting this brew for a year I think I'm fine with anything as long as it is documented so the user doesn't run into unexpected troubles. Having it configurable seems like a perfect solution to me

Reply:
I think now that the warning I sketched above is superfluous as mountpoint-related vfs stuff is behind a mutex anyway (but please verify, I'm on mobile and can't easily read long code here)